spec/integration/util/windows/registry_spec.rb

```diff
@@ -163,5 +163,75 @@ def expects_registry_value(array)
         end
       end
     end
+
+    context "when reading binary values" do
+      let (:hklm) { Win32::Registry::HKEY_LOCAL_MACHINE }
+      let (:puppet_key) { "SOFTWARE\\Puppet Labs"}
+      let (:subkey_name) { "PuppetRegistryTest" }
+      let (:guid) { SecureRandom.uuid }
+
+      before :each do
+        hklm.create("#{puppet_key}\\#{subkey_name}", Win32::Registry::KEY_ALL_ACCESS) do |reg|
+          reg.write("#{guid}", Win32::Registry::REG_BINARY, "abinarystring")
+        end
+      end
+
+      after(:each) do
+        hklm.open(puppet_key, Win32::Registry::KEY_ALL_ACCESS) do |reg|
+          subject.delete_key(reg, subkey_name)
+        end
+      end
+
+      it "should return binary values" do
+
+        hklm.open("#{puppet_key}\\#{subkey_name}", Win32::Registry::KEY_ALL_ACCESS) do |reg|
```

Comment:
Probably not. As it was just the spec tests I did not see the need to find the precise permissions required.  I will investigate other options.